src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -286,6 +286,49 @@ async Task<GlobalNotificationState> continuation(Task<GlobalNotificationState> p
             }
         }
 
+        #endregion
+
+        #region Assets
+
+        /// <summary>
+        /// Remote API.
+        /// </summary>
+        public async Task GetAssetsAsync(int scopeId, Checksum[] checksums, string pipeName, CancellationToken cancellationToken)
+        {
+            try
+            {
+                using (Logger.LogBlock(FunctionId.JsonRpcSession_RequestAssetAsync, pipeName, cancellationToken))
+                {
+                    await RemoteEndPoint.WriteDataToNamedPipeAsync(
+                        pipeName,
+                        (scopeId, checksums),
+                        (writer, data, cancellationToken) => RemoteHostAssetSerialization.WriteDataAsync(writer, RemotableDataService, data.scopeId, data.checksums, cancellationToken),
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex) when (FatalError.ReportWithoutCrashUnlessCanceledAndPropagate(ex, cancellationToken))
+            {
+                throw ExceptionUtilities.Unreachable;
+            }
+        }
+
+        /// <summary>
+        /// Remote API.
+        /// </summary>
+        public Task<bool> IsExperimentEnabledAsync(string experimentName, CancellationToken cancellationToken)
+        {
+            try
+            {
+                return Task.FromResult(Workspace.Services.GetRequiredService<IExperimentationService>().IsExperimentEnabled(experimentName));
+            }
+            catch (Exception ex) when (FatalError.ReportWithoutCrashUnlessCanceledAndPropagate(ex, cancellationToken))
```

Comment:
do we hit this exception?  seems like it would be very unfortunate for the aove to every fail.

Reply:
We better not hit any exception like this - that's why we're reporting NFW if we do :), but anything is possible. Note that this may call to VS API. 

Generally, we want to report NFW on remote call boundaries before we leave the process, otherwise we might fail in the other process without having full stack of the root cause.